### PR TITLE
Factor `getAccountBalance` into `getCachedAccountBalance` and `fetchAccountBalances`

### PR DESCRIPTION
**Pull request by Anviking**
# Issue Number

Split off from #2684 

# Overview

- [x] Rename `getAccountBalance` to `getCachedAccountBalance` for clarity
- [x] Add `fetchAccountBalances` function for un-cached behaviour
- [x] `fetchAccountBalances` from `listStakeKeys`  

# Comments

### CODE REVIEW

File: lib/core/src/Cardano/Wallet/Network.hs [hunk 1]
```diff
@@ -172,9 +175,20 @@ data NetworkLayer m block = NetworkLayer
...
         :: RewardAccount
-        -> ExceptT ErrGetAccountBalance m Coin
+        -> m Coin
+        -- ^ Return the cached reward balance of an account.
+        --
+        -- If there is no cached value, it will return `Coin 0`, and add the
+        -- account to the internal set of observed account, such that it will be
+        -- fetched later.
+
+    , fetchAccountBalances
```
Comment (paweljakubas) [inline lib/core/src/Cardano/Wallet/Network.hs]: occasionally I think that  `fetchRewardAccountBalances` and `getCachedRewardAccountBalance` would be better here  in spite of being long...

**Review by paweljakubas — APPROVED** [review]
I like this decomposition. If we have a proof that it helps with flaky integration testing then it is great

**Comment by Anviking** [comment]
Thanks for having a look.

> If we have a proof that it helps with flaky integration testing then it is great

This wasn't intended to fix flaky tests, although that's a good point, this caching-behaviour could be involved 🤔 

(This PR only changes the behaviour of listStakeKeys. Wallet reward balances still use the cached version.)

File: lib/core/src/Cardano/Wallet/Network.hs (hunk 1 above)
Comment (Anviking) [reply lib/core/src/Cardano/Wallet/Network.hs]: 👍 fair -> d81089b8bb205af2f082d0704eac7410aba1c541

**Pull request merged**